### ISSUE TITLE
[15.0] Odoo is not compatible with Python 3.10

### ISSUE THREAD

**Issue by frederictobiasc** [odoo/odoo#84958]
The use of the Alias [collections.Set](https://github.com/odoo/odoo/blob/15.0/odoo/models.py#L6598) is deprecated since python 3.3 (Sept. 29, 2012).

Impacted versions: 15.0

Steps to reproduce:
1. Setup a server running python 3.10
2. Install Odoo
3. Start Odoo

Current behavior:
```
Feb 21 06:52:28 nb-istobic systemd[1]: Started Odoo Open Source ERP and CRM.
Feb 21 06:52:29 nb-istobic odoo[111364]: Traceback (most recent call last):
Feb 21 06:52:29 nb-istobic odoo[111364]:   File "/usr/bin/odoo", line 5, in <module>
Feb 21 06:52:29 nb-istobic odoo[111364]:     import odoo
Feb 21 06:52:29 nb-istobic odoo[111364]:   File "/usr/lib/python3.10/site-packages/odoo/__init__.py", line 115, in <module>
Feb 21 06:52:29 nb-istobic odoo[111364]:     from . import osv
Feb 21 06:52:29 nb-istobic odoo[111364]:   File "/usr/lib/python3.10/site-packages/odoo/osv/__init__.py", line 4, in <module>
Feb 21 06:52:29 nb-istobic odoo[111364]:     from . import osv
Feb 21 06:52:29 nb-istobic odoo[111364]:   File "/usr/lib/python3.10/site-packages/odoo/osv/osv.py", line 5, in <module>
Feb 21 06:52:29 nb-istobic odoo[111364]:     from ..models import Model, TransientModel, AbstractModel
Feb 21 06:52:29 nb-istobic odoo[111364]:   File "/usr/lib/python3.10/site-packages/odoo/models.py", line 6590, in <module>
Feb 21 06:52:29 nb-istobic odoo[111364]:     collections.Set.register(BaseModel)
Feb 21 06:52:29 nb-istobic odoo[111364]: AttributeError: module 'collections' has no attribute 'Set'
```

Expected behavior: Normal start

Sources:
https://bugs.python.org/issue37324
https://github.com/python/cpython/commit/c47c78b878ff617164b2b94ff711a6103e781753

**Comment by frederictobiasc** [comment]
https://github.com/odoo/odoo/pull/81918
https://github.com/odoo/odoo/pull/60623/

**Comment by frederictobiasc** [comment]
I suppose this is fixed now? #85969

**Comment by mart-e** [comment]
Yes should be, I tried to convert the ones I found. Don't hesitate to open a PR if there is still other issues on this :)
